core/src/allocation.jl

```diff
@@ -0,0 +1,531 @@
+            for allocgraph_edge_composite in allocgraph_edges_composite
+                if allocgraph_edge_composite[1] in subnetwork_neighbor_ids
+                    pushfirst!(allocgraph_edge_composite, subnetwork_node_id)
+                    found_edge = true
+                    break
+                elseif allocgraph_edge_composite[end] in subnetwork_neighbor_ids
+                    push!(allocgraph_edge_composite, subnetwork_node_id)
+                    found_edge = true
+                    break
+                end
+            end
+
+            # Start a new allocgraph composite edge if no existing edge to append to was found
+            if !found_edge
+                push!(allocgraph_edges_composite, [subnetwork_node_id])
+            end
+        end
+    end
+
+    # For the composite allocgraph edges:
+    # - Find out whether they are connected to allocgraph nodes on both ends
+    # - Compute their capacity
+    # - Find out their allowed flow direction(s)
+    for allocgraph_edge_composite in allocgraph_edges_composite
+        # Find allocgraph node connected to this edge on the first end
+        allocgraph_node_id_1 = nothing
+        subnetwork_neighbors_side_1 =
+            all_neighbors(graph_flow, allocgraph_edge_composite[1])
+        for subnetwork_neighbor_node_id in subnetwork_neighbors_side_1
+            if subnetwork_neighbor_node_id in subnetwork_node_ids_represented
+                allocgraph_node_id_1 = node_id_mapping[subnetwork_neighbor_node_id][1]
+                pushfirst!(allocgraph_edge_composite, subnetwork_neighbor_node_id)
+                break
+            end
+        end
+
+        # No connection to a max flow node found on this side, so edge is discarded
+        if isnothing(allocgraph_node_id_1)
+            continue
+        end
+
+        # Find allocgraph node connected to this edge on the second end
+        allocgraph_node_id_2 = nothing
+        subnetwork_neighbors_side_2 =
+            all_neighbors(graph_flow, allocgraph_edge_composite[end])
+        for subnetwork_neighbor_node_id in subnetwork_neighbors_side_2
+            if subnetwork_neighbor_node_id in subnetwork_node_ids_represented
+                allocgraph_node_id_2 = node_id_mapping[subnetwork_neighbor_node_id][1]
+                # Make sure this allocgraph node is distinct from the other one
+                if allocgraph_node_id_2 ≠ allocgraph_node_id_1
+                    push!(allocgraph_edge_composite, subnetwork_neighbor_node_id)
+                    break
+                end
+            end
+        end
+
+        # No connection to allocgraph node found on this side, so edge is discarded
+        if isnothing(allocgraph_node_id_2)
+            continue
+        end
+
+        # Find capacity of this composite allocgraph edge
+        positive_flow = true
+        negative_flow = true
+        allocgraph_edge_capacity = Inf
+        for (i, subnetwork_node_id) in enumerate(allocgraph_edge_composite)
```

Comment:
I think iterating over a partition of three elements makes more sense here: https://docs.julialang.org/en/v1/base/iterators/#Base.Iterators.partition

Using raw indexes should only be seen as the last option after functions in the [Iteration utilities](https://docs.julialang.org/en/v1/base/iterators/#Iteration-utilities) have been exhausted.